src/PrestaShopBundle/Form/Admin/Configure/ShopParameters/TrafficSeo/Meta/MetaType.php

```diff
@@ -0,0 +1,183 @@
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Form\Admin\Configure\ShopParameters\TrafficSeo\Meta;
+
+use PrestaShop\PrestaShop\Core\ConstraintValidator\Constraints\DefaultLanguage;
+use PrestaShop\PrestaShop\Core\ConstraintValidator\Constraints\IsUrlRewrite;
+use PrestaShopBundle\Form\Admin\Type\TranslatableType;
+use PrestaShopBundle\Translation\TranslatorAwareTrait;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
+use Symfony\Component\Validator\Constraints\NotBlank;
+use Symfony\Component\Validator\Constraints\Regex;
+
+/**
+ * Class MetaType is responsible for providing form fields for Shop parameters -> Traffic & Seo ->
+ * Seo & Urls -> add and edit forms.
+ */
+class MetaType extends AbstractType
+{
+    use TranslatorAwareTrait;
+
+    /**
+     * @var array
+     */
+    private $defaultPageChoices;
+
+    /**
+     * @var array
+     */
+    private $modulePageChoices;
+
+    /**
+     * @param array $defaultPageChoices
+     * @param array $modulePageChoices
+     */
+    public function __construct(
+        array $defaultPageChoices,
+        array $modulePageChoices
+    ) {
+        $this->defaultPageChoices = $defaultPageChoices;
+        $this->modulePageChoices = $modulePageChoices;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder
+            ->add('page_name', ChoiceType::class, [
+                'choices' => [
+                    $this->trans('Default pages', [], 'Admin.Shopparameters.Feature') => $this->defaultPageChoices,
+                    $this->trans('Module pages', [], 'Admin.Shopparameters.Feature') => $this->modulePageChoices,
+                ],
+                'constraints' => [
+                    new NotBlank([
+                        'message' => $this->trans(
+                            'The %s field is required.',
+                            [
+                                sprintf('"%s"', $this->trans('Page name', [], 'Admin.Shopparameters.Feature')),
+                            ],
+                            'Admin.Notifications.Error'
+                        ),
+                    ]),
+                    new Regex([
+                        'pattern' => '/^[a-zA-Z0-9_.-]+$/',
+                        'message' => $this->trans(
+                            '%s is invalid.',
+                            [],
```

Comment:
@tomas862 What value %s will get when no options are passed in this case? Will it be translatable?

Reply:
it auto fetches the incorrect input value so if the option "incorrect option" is selected then it will display : "incorrect option" is invalid.